### PR TITLE
Parquet file supports for datasets

### PR DESCRIPTION
**Pull request by koropets**

### CODE REVIEW

File: gordo/machine/dataset/data_provider/ncs_reader.py [hunk 1]
```diff
@@ -1,21 +1,99 @@
...
+
+    def __init__(self):
+        header = ["Sensor", "Value", "Time", "Status"]
+        super().__init__(CsvFileType(header, time_series_columns))
+
+    def lookup(
+        self, client: core.AzureDLFileSystem, dir_path: str, tag_name: str, year: int
+    ) -> Optional[str]:
+        file_extension = self.file_type.file_extension
+        path = f"{dir_path}/{tag_name}_{year}{file_extension}"
```
Comment (hjarraya) [inline gordo/machine/dataset/data_provider/ncs_reader.py:77]: It looks like the path is slightly different than what is supposed to be, before your change the file path was
`f"{tag_base_path}/{tag_name_encoded}/{tag_name_encoded}_{year}.csv"` 
Where `tag_name_encoded = quote(tag.name, safe=" ")`

File: gordo/machine/dataset/data_provider/ncs_reader.py [hunk 2]
```diff
@@ -1,21 +1,99 @@
...
+    """
+
+    def __init__(self):
+        super().__init__(ParquetFileType(time_series_columns))
+
+    def lookup(
+        self, client: core.AzureDLFileSystem, dir_path: str, tag_name: str, year: int
+    ) -> Optional[str]:
+        file_extension = self.file_type.file_extension
+        path = f"{dir_path}/parquet/{tag_name}_{year}{file_extension}"
```
Comment (hjarraya) [inline gordo/machine/dataset/data_provider/ncs_reader.py:93]: `tag_name` needs encoding probably.

File: gordo/machine/dataset/data_provider/ncs_reader.py (hunk 1 above)
Comment (koropets) [reply gordo/machine/dataset/data_provider/ncs_reader.py:77]: For this function, I am passing already encoded `tag_name` https://github.com/equinor/gordo/blob/parquet_file_format/gordo/machine/dataset/data_provider/ncs_reader.py#L331 . I have mentioned this [here](https://github.com/equinor/gordo/blob/parquet_file_format/gordo/machine/dataset/data_provider/ncs_reader.py#L57)

**Review by hjarraya — APPROVED** [review]
LGTM

**Pull request merged**